querybook/webapp/components/Search/SearchOverview.tsx

```diff
@@ -320,73 +367,263 @@ export const SearchOverview: React.FunctionComponent = () => {
+    const queryEngineFilterDOM = (
+        <div className="filter-query-engine">
+            <ReactSelect
+                styles={defaultReactSelectStyles}
+                value={
+                    searchFilters['engine_id'] && {
+                        label: queryEngineById[searchFilters['engine_id']].name,
+                        value: searchFilters['engine_id'],
+                    }
+                }
+                onChange={(option) => {
+                    const engineId = option?.value;
+                    updateSearchFilter('engine_id', engineId);
+                }}
+                options={queryEngines.map((engine) => ({
+                    label: engine.name,
+                    value: engine.id,
+                }))}
+                isClearable
+            />
+        </div>
+    );
+
+    const queryTypeFilterDOM = (
+        <div className="filter-query-type">
+            <ReactSelect
+                styles={defaultReactSelectStyles}
+                value={
+                    searchFilters['query_type'] && {
+                        label: titleize(searchFilters['query_type'], '_', ' '),
+                        value: searchFilters['query_type'],
+                    }
+                }
+                onChange={(option) =>
+                    updateSearchFilter('query_type', option?.value)
+                }
+                options={['query_cell', 'query_execution'].map((queryType) => ({
+                    label: titleize(queryType, '_', ' '),
+                    value: queryType,
+                }))}
+                isClearable
+            />
+        </div>
+    );
+
+    const statementTypeFilterDOM = (
+        <div className="filter-statement-type">
+            <ReactSelect
+                styles={defaultReactSelectStyles}
+                value={
+                    searchFilters['statement_type']
+                        ? searchFilters['statement_type'].map((statement) => ({
+                              label: statement.toUpperCase(),
+                              value: statement,
+                          }))
+                        : null
+                }
+                onChange={(values) => {
+                    const statementTypes =
+                        values && values.length
+                            ? values.map(({ value }) => value)
+                            : null;
+                    updateSearchFilter('statement_type', statementTypes);
+                }}
+                options={['SELECT', 'INSERT', 'CREATE', 'DELETE'].map(
+                    (statement) => ({
+                        label: statement,
+                        value: statement,
+                    })
+                )}
+                isMulti
+                isClearable
+            />
+        </div>
+    );
+
+    const tableFilterDOM = (
+        <TableSelect
+            tableNames={searchFilters['full_table_name'] || []}
+            onSelect={(tableName) => {
```

Comment:
let TableSelect manage the state of tableNames, so onSelect can just be `(newTableNames: string[]) => updateSearchFilter('full_table_name', newTableNames)`

Reply:
updated to `setTableNames` so that the TableSelect component can also use that in `onRemove`